src/firestarter/Environment/X86/Payload/AVX512_AMX_Payload.cpp

```diff
@@ -0,0 +1,592 @@
+  }
+  cb.mov(l1_addr, pointer_reg);
+
+  if (dumpRegisters) {
+    auto SkipRegistersDump = cb.newLabel();
+
+    cb.test(ptr_64(pointer_reg, -8), Imm(firestarter::DumpVariable::Wait));
+    cb.jnz(SkipRegistersDump);
+
+    // dump all the ymm register
+    for (int i = 0; i < (int)this->registerCount(); i++) {
+      cb.vmovapd(
+          zmmword_ptr(pointer_reg, -64 - this->registerSize() * 8 * (i + 1)),
+          Zmm(i));
+    }
+
+    // set read flag
+    cb.mov(ptr_64(pointer_reg, -8), Imm(firestarter::DumpVariable::Wait));
+
+    cb.bind(SkipRegistersDump);
+  }
+
+  if (errorDetection) {
+    this->emitErrorDetectionCode<decltype(iter_reg), Zmm>(
+        cb, iter_reg, addrHigh_reg, pointer_reg, temp_reg, temp_reg2);
+  }
+
+  cb.test(ptr_64(addrHigh_reg), Imm(LOAD_HIGH));
+  cb.jnz(Loop);
+
+  cb.bind(FunctionExit);
+
+  cb.movq(rax, iter_reg);
+
+  cb.emitEpilog(frame);
+
+  cb.finalize();
+
+  // String sb;
+  // cb.dump(sb);
+
+  Error err = this->rt.add(&this->loadFunction, &code);
+  if (err) {
+    workerLog::error() << "Asmjit adding Assembler to JitRuntime failed in "
+                       << __FILE__ << " at " << __LINE__;
+    return EXIT_FAILURE;
+  }
+
+  // skip if we could not determine cache size
+  if (l1i_cache_size != 0) {
+    auto loopSize = code.labelOffset(FunctionExit) - code.labelOffset(Loop);
+    auto instructionCachePercentage = 100 * loopSize / l1i_cache_size;
+
+    if (loopSize > l1i_cache_size) {
+      workerLog::warn() << "Work-loop is bigger than the L1i-Cache.";
+    }
+
+    workerLog::trace() << "Using " << loopSize << " of " << l1i_cache_size
+                       << " Bytes (" << instructionCachePercentage
+                       << "%) from the L1i-Cache for the work-loop.";
+    workerLog::trace() << "Sequence size: " << sequence.size();
+    workerLog::trace() << "Repetition count: " << repetitions;
+  }
+
+  return EXIT_SUCCESS;
+}
+
+std::list<std::string> AVX512_AMX_Payload::getAvailableInstructions() const {
+  std::list<std::string> instructions;
+
+  transform(this->instructionFlops.begin(), this->instructionFlops.end(),
+            back_inserter(instructions),
+            [](const auto &item) { return item.first; });
+
+  return instructions;
+}
+
+void AVX512_AMX_Payload::init(unsigned long long *memoryAddr,
+                         unsigned long long bufferSize) {
+  X86Payload::init(memoryAddr, bufferSize, 0.27948995982e-4, 0.27948995982e-4);
+}
+
+void create_AMX_config(__tilecfg *tileinfo){
+  // Create tile_cfg, fill it and return 
+
+  int i;
+  tileinfo->palette_id = 1;
+  tileinfo->start_row = 0;
+
+
+  for (i = 0; i < 8; ++i)
+  {
+    tileinfo->colsb[i] = MAX_COLS;
+    tileinfo->rows[i] =  MAX_ROWS;
+  }
+
+  _tile_loadconfig(tileinfo);
+}
+
+
+void request_permission(){
+
+  long rc;
+  unsigned long bitmask;
+  rc = syscall(SYS_arch_prctl, ARCH_REQ_XCOMP_PERM, XFEATURE_XTILEDATA);
+
+  if(rc){
+  	printf("XTILE_DATA request failed: %ld", rc);
```

Comment:
please change to workerLog:: trace, error etc.

Reply:
Done